tasks/README.md

```diff
@@ -98,9 +103,17 @@ Any parameter that does not specify a type will accept any value (default type i
 
 If a parameter type accepts `null` the task runner will accept either a `null` value or the absence of the property. Task authors must accept missing properties for nullable parameters. Task authors must not differentiate between absent properties and properties with `null` values.
 
+### Metaparameters
+
+In addition to the tasks parameters, the task runner may inject metaparameters prefixed by '_'. These include `_noop` and `_task`.
+
 ## Task execution
 
-The task file is copied to the target and then executed on the target by the task runner.
+If the task has multiple executable files, the `implementations` field of the metadata is used to determine which executable is suitable for the target. Each implementation can specify `requirements`, which is an array of the required "features" to use that implementation. The available features are defined by the task runner.
```

Comment:
We should document the features that task runners should provide.

We should document that task authors should name their tasks to supports backwards compatibility.

Reply:
Do you think that latter point belongs in the spec or in the writing tasks doc?